### PR TITLE
NO-TICKET: refactoring referring domain handling

### PR DESCRIPTION
**Pull request by HeikoKampe**
- fix missing assignment of s._referringDomain
- moved function getDomainFromURLString() to new object 'utils'
- improved handling of refferingDomain in channels functionality

### CODE REVIEW

File: extensions/doPlugins_global.js [hunk 1]
```diff
@@ -27,6 +27,18 @@ s.split = new Function("l","d",""
...
+    getDomainFromURLString: function (urlString) {
+        try {
+            const urlObject = new URL(urlString);
+            return urlObject.hostname;
+        } catch (err) {
+            return '';
+        }
+    }
+};
+
```
Comment (mehakraza) [inline extensions/doPlugins_global.js:41]: utils 👍

File: extensions/doPlugins_global.js [hunk 2]
```diff
@@ -378,16 +379,16 @@ function init() {
...
     //height & width for iPhones
     if (window.navigator.userAgent.indexOf('iPhone') > -1) {
         s.eVar94 = window.screen.width + 'x' + window.screen.height;
     }
 
+    campaign.setCampaignVariables(s);
+    setPageSourceForCheckout(s);
+    setExternalReferringDomainEvents(s);
     ICIDTracking.setVariables(s);
     campaign.setCampaignVariables(s);
```
Comment (mehakraza) [inline extensions/doPlugins_global.js:393]: setCampaignVariables seems to be called twice (Line 389 and 393).

**Review by mehakraza — APPROVED** [review]
Good to merge 👍

File: extensions/doPlugins_global.js [hunk 3]
```diff
@@ -76,26 +79,23 @@ const articleViewType = {
         });
     },
 
-    isFromBild: function (referrer) {
-        const referrerDomain = this.getDomainFromURLString(referrer);
+    isFromBild: function (referrerDomain) {
```
[the remark this answers is not in the document]
Comment (HeikoKampe) [reply extensions/doPlugins_global.js]: fixed

File: extensions/doPlugins_global.js (hunk 2 above)
Comment (HeikoKampe) [reply extensions/doPlugins_global.js:392]: good catch! fixed

**Pull request merged**